fpga_interchange/xc7_device_timing_patching.py

```diff
@@ -0,0 +1,246 @@
+                    cap = float(model_values['in_cap'])
+                res = 0
+                if model_values['res'] is not None:
+                    res = float(model_values['res'])
+                pip_models[pip] = (cap, res, delays)
+            for site in tile_data['sites']:
+                siteType = siteName_siteType_map[string_map[site['type']]]
+                for sitePin, dic in site['site_pins'].items():
+                    sitePin_obj = siteType_name_sitePin_map[(
+                        siteType, string_map[sitePin])]
+                    if dic is None:
+                        dic = {}
+                        dic['delay'] = [0, 0, 0, 0]
+                    elif 'res' in dic.keys():
+                        sitePin_obj.model.init('resistance')
+                        sitePin_obj.model.resistance.slow.init('slow')
+                        sitePin_obj.model.resistance.slow.slow.typ.typ = float(
+                            dic['res']) * 1e-6
+                    else:
+                        sitePin_obj.model.init('capacitance')
+                        sitePin_obj.model.capacitance.slow.init('slow')
+                        sitePin_obj.model.capacitance.slow.slow.typ.typ = float(
+                            dic['cap']) * 1e-6
+                    sitePin_obj.init('delay')
+                    corner_model = sitePin_obj.delay
+                    for i, delay in enumerate(dic['delay']):
+                        delay = float(delay) * 1e-9
+                        attr1 = "fast" if i < 2 else "slow"
+                        attr2 = "min" if i % 2 == 0 else "max"
+                        v = getattr(corner_model, attr1)
+                        if v.which() != attr1:
+                            v.init(attr1)
+                        v = getattr(v, attr1)
+                        v = getattr(v, attr2)
+                        setattr(v, attr2, delay)
+
+    timing_set = set()
+    for timing in node_model_map.values():
+        timing_set.add(timing)
+    timing_dict = {timing: i for i, timing in enumerate(timing_set)}
+    dev.init("nodeTimings", len(timing_dict))
+    for model, i in timing_dict.items():
+        corner_model = dev.nodeTimings[i]
+        corner_model.capacitance.slow.init("slow")
+        corner_model.capacitance.slow.slow.typ.typ = model[0]
+        corner_model.resistance.slow.init("slow")
+        corner_model.resistance.slow.slow.typ.typ = model[1]
+
+    for node, timing in node_model_map.items():
+        node.nodeTiming = timing_dict[timing]
+
+    timing_set = set()
+    for model in pip_models.values():
+        timing_set.add(model)
+    timing_dict = {timing: i for i, timing in enumerate(timing_set)}
+    dev.init("pipTimings", len(timing_dict))
+    for model, i in timing_dict.items():
+        pipTiming = dev.pipTimings[i]
+        corner_model = pipTiming.internalCapacitance
+        corner_model.slow.init("slow")
+        corner_model.slow.slow.typ.typ = 0
```

Comment:
I think the capacitance value in prjxray-db corresponds to "internal capacitance" in the interchange schema: https://github.com/SymbiFlow/prjxray/blob/cb5f2a30633bf0d3ac91a9b5802790ac1da86736/prjxray/timing.py#L29; capacitance should be allocated to `internalCapacitance` and `inputCapacitance` set to zero instead

Reply:
yeah, my bad, it will be fixed in next commit